### PR TITLE
Replace PF3 Alerts

### PR DESCRIPTION
**Pull request by rszwajko**
This PR depends on #1525 

Replace existing PF3 Alerts with newer PF4 widgets.
Key points:
1. PF4 requires a title (which makes sens from usability point of view). However some our alerts have no title which makes them look bad after migration. Where possible a generic title was added i.e. "Messages" or "Details"
2. PF4 styling favors displaying alerts above the text (i.e. bold line at the top of the alert box). In one case (error on VM Details screen) we display alert at the bottom.
3. PF4 introduces self-dismissing alerts which makes our CounterAlert obsolete. However for now the original design was used.
4. in PF4 ToastNotifications are also plain Alerts ~~- this will be handled in separate PR.~~

### CODE REVIEW

**Comment by rszwajko** [comment]
### Screenshots for toast notifications
#### existing PF3 
toast_notifications_PF3
#### new PF4 notification (with title)
toast_notifications_PF4
#### new PF4 notification with generic title. 
The generic title is just a fallback for messages that provide no title. Missing title can be added later on in a follow up PR (in some cases this requires splitting the message into 2 parts which will invalidate translations).
toast_notificattions_PF4_with_generic_title

**Comment by rszwajko** [comment]
Note that alerts in "Create snapshot" modal and "VM has pending configuration..." can be replaced with [Hint component](https://www.patternfly.org/v4/components/hint).
See PR #1540  for new screenshots.

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by ovirt-infra** [comment]
All tests passed

File: src/components/CounterAlert.js [hunk 1]
```diff
@@ -45,10 +45,14 @@ class CounterAlert extends React.Component {
   render () {
     const { title, type, children } = this.props
     return this.state.showAlert && (
-      <Alert type={type} onDismiss={this.handleDismiss} className={style['text-align-left']}>
-        {title}
-        {children}
-      </Alert>
+      <Alert
+        variant={type}
+        onTimeout={this.handleDismiss}
```
Comment (sjd78) [inline src/components/CounterAlert.js:50]: `onTimeout` does not appear to be needed

File: src/components/CreateVmWizard/steps/SummaryReview.js [hunk 2]
```diff
@@ -290,13 +291,13 @@ const SummaryReview = ({
...
           </div>
           { progress.messages && progress.messages.length > 0 && (
-            <Alert type='success' >
-              {progress.messages.map((message, index) => (
-                <div key={`message-${index}`}>
-                  { message }
-                </div>
-              )
-              )}
+            <Alert variant='success' isInline title={msg.messages()}>{
```
Comment (sjd78) [inline src/components/CreateVmWizard/steps/SummaryReview.js:294]: "Messages" is ok here for the title, but seems a bit awkward.  I have no better suggestion.  At least it is not likely to be displayed!

File: src/components/ToastNotifications.js [hunk 3]
```diff
@@ -2,36 +2,68 @@ import React from 'react'
...
+  }
 }
 
 const ToastNotifications = ({ userMessages, onDismissNotification, msg }) => {
   return (
-    <ToastNotificationList>
-      { userMessages.get('records').filter(r => !r.get('notified')).map(r => (
-        <TimedToastNotification
+    <AlertGroup isToast isLiveRegion>
+      { userMessages.get('records').toJS().filter(({ notified }) => !notified).map(r => (
```
Comment (sjd78) [inline src/components/ToastNotifications.js:48]: `toJS()` is nice -- a little less ImmutableJS is a good thing!

**Review by sjd78 — COMMENTED** [review]
Only a few comments, but looks pretty good.

**Comment by ovirt-infra** [comment]
All tests passed

**Comment by sgratch** [comment]
@sjd78 @rszwajko need to open another dup PR for that since we do want to merge it later....

**Pull request merged**